### PR TITLE
Update all the tests

### PR DESCRIPTION
**Pull request by folkertdev**
makes all but one test run. I'll add some inline comments with thoughts

### CODE REVIEW

File: tests/all/test_instruction_values.rs [hunk 1]
```diff
@@ -4,6 +4,7 @@ use inkwell::values::{BasicValue, InstructionOpcode::*};
 use inkwell::{AddressSpace, AtomicOrdering, AtomicRMWBinOp, FloatPredicate, IntPredicate};
 
 #[test]
+#[ignore]
```
Comment (folkertdev) [inline tests/all/test_instruction_values.rs:7]: this is the ignored test. Something changed with the operands of the free instruction. I'm not quite sure what. The problem is that `free_operand0` no longer has an instruction value associated with it.

File: src/builder.rs [hunk 2]
```diff
@@ -213,30 +216,59 @@ impl<'ctx> Builder<'ctx> {
...
+        &self,
+        function: FunctionValue<'ctx>,
+        args: &[BasicMetadataValueEnum<'ctx>],
+        name: &str,
+    ) -> CallSiteValue<'ctx> {
+        self.build_call_help(function.get_type(), function.as_value_ref(), args, name)
+    }
 
+    #[llvm_versions(15.0..=latest)]
+    pub fn build_indirect_call(
```
Comment (hedgar2017) [inline src/builder.rs:278]: Do I understand correctly that this function can be used everywhere, for both global function values and generic function pointers?

**Review by hedgar2017 — APPROVED** [review]

**Review by hedgar2017 — CHANGES_REQUESTED** [review]
Please fix v5 and we can address that ignored test problem is another PR.

File: src/builder.rs (hunk 2 above)
Comment (folkertdev) [reply src/builder.rs:278]: you could turn a function into a global pointer and then use that with this function, yes. But then you have to supply the function type explicitly.

Comment (hedgar2017) [reply src/builder.rs:278]: Cool, that's what I need.

**Comment by hedgar2017** [comment]
@folkertdev could you maybe add doc tests for the direct and indirect calls?

**Comment by folkertdev** [comment]
should be all good now. llvm 4 + 5 are fixed and documentation examples were added.

**Review by hedgar2017 — APPROVED** [review]

**Pull request merged**